File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>   

int main(void)
{
    printf("Oh hey! How much change is owed?\n");
    float owed;
    owed = get_float();
    while (owed<=0)
        {
        printf("How much change is owed?\n");
        owed = get_float();
        }
    int owedc = (owed+.005)*100;
    int coinsused=0;
    while(owedc >= 25)
    {
        owedc = owedc - 25;
        coinsused++;
    }    
    while(owedc >= 10)
    {
        owedc = owedc - 10;
        coinsused++;
    }
    while(owedc >= 5)
    {
        owedc = owedc - 5;
        coinsused++;
    }
    while(owedc >= 1)
    {
        owedc = owedc - 1;
        coinsused++;
    }
    printf("%i\n",coinsused);
}

File: pset6/greedy.py
import cs50

def main():
    while True:
        print("Please indicate how much change is owed.")
        c = cs50.get_float()
        if c > 0:
            break
    get_greedy(c)
    
def get_greedy(c):
    qtr = 0.25
    dime = 0.10
    nick = 0.5
    pen = 0.01
    cu = 0
    while c >= qtr:
        c = c - qtr
        cu += 1
    while c >= dime:
        c = c - dime
        cu += 1    
    while c >= nick:
        c = c - nick
        cu += 1    
    while c >= pen:
        c = c - pen
        cu += 1
    print("{}".format(cu))
        
if __name__ == "__main__":
    main()